plugins/woocommerce/client/blocks/assets/js/blocks/product-description/edit.tsx

```diff
@@ -1,83 +1,29 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { useQueryLoopProductContextValidation } from '@woocommerce/base-hooks';
 import {
-	useBlockProps,
-	useInnerBlocksProps,
 	RecursionProvider,
+	useBlockProps,
 	useHasRecursion,
 	Warning,
-	__experimentalUseBlockPreview as useBlockPreview,
 } from '@wordpress/block-editor';
-import { parse } from '@wordpress/blocks';
-import {
-	useEntityProp,
-	useEntityBlockEditor,
-	store as coreStore,
-} from '@wordpress/core-data';
-import { useSelect } from '@wordpress/data';
-import { useMemo } from '@wordpress/element';
-import { useQueryLoopProductContextValidation } from '@woocommerce/base-hooks';
+import { useEntityProp } from '@wordpress/core-data';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import { ProductDescriptionEditProps } from './types';
 
-/**
- * Returns whether the current user can edit the given entity.
- *
- * @param {string} kind     Entity kind.
- * @param {string} name     Entity name.
- * @param {string} recordId Record's id.
- */
-export function useCanEditEntity( kind, name, recordId ) {
-	return useSelect(
-		( select ) =>
-			select( coreStore ).canUser( 'update', {
-				kind,
-				name,
-				id: recordId,
-			} ),
-		[ kind, name, recordId ]
-	);
-}
-
-function ReadOnlyContent( {
-	parentLayout,
-	layoutClassNames,
-	userCanEdit,
-	postType,
-	postId,
-} ) {
+function ReadOnlyContent( { layoutClassNames, postType, postId } ) {
```

Comment:
For sake of simplicity what about removing `ReadOnlyContent` completely and moving its body to `Content` given that `Content` only renders `ReadOnlyContent` now.